### ISSUE TITLE
JSON API support

### ISSUE THREAD

**Issue by Baukereg** [lytics/ember-data-model-fragments#196]
Are there plans to include support for JSON API?

So far, `array` seems to work, but I run into trouble with `fragment` and `fragmentArray`.

```
// models/person
import DS from 'ember-data';
import { fragment } from 'model-fragments/attributes';

export default DS.Model.extend({

  name: fragment('personName')
  // or person-name, doesn't seem to matter
  
});
```

```
// models/person-name
import DS from 'ember-data';
import Fragment from 'model-fragments/fragment';

export default Fragment.extend({

  fullName: DS.attr('string')

});
```

```
// server payload
{
  'data': {
    'type': 'person',
    'id': 'person_self',
    'attributes': {
      'name': {
        'full-name': 'Dart Vader'
      }
    }
  }
}
```

```
// in person router
console.log(model.get('name.fullName')); //undefined

**Comment by slindberg** [comment]
TL;DR Create a serializer for your fragments that understands dasherized attribute names.

First take a look at the [serializing section of the readme](https://github.com/lytics/ember-data-model-fragments#serializing). ED uses transforms to deserialize attribute values, and MF adds transforms for fragments and fragment arrays. These transforms use fragment-specific serializers to deserialize those nested JSON attribute values. This is so that you can have a `NameSerializer` that provides custom logic for dealing with serializing/deserializing `Name` models, and not have to jam it into the parent serializer.

Here's the rub: `JSONAPISerializer` deals with the top level structure of your API responses, and looks for a bunch of specific keys that don't make sense in the context of a nested JSON fragment. So fragment serializers inherit from `JSONSerializer` instead, which doesn't understand dasherized attributes. What you need to do is create a fragment serializer that overrides [`keyForAttribute`](http://emberjs.com/api/data/classes/DS.JSONSerializer.html#method_keyForAttribute), and register it in the container under the `serializer:-fragment` key.

**Comment by fran-worley** [comment]
@slindberg Given that ember-data default is JSON-API wouldn't it make sense for this to be supported 'out of the box'? or at least have a complete working example. I've just run into this and was surprised that it didn't just work.

**Comment by slindberg** [comment]
cc @jakesjews @workmanw

**Comment by fran-worley** [comment]
For anyone else who comes here, this is the simplest application fragment serialiser to get things working:

```javascript
// app/serializers/fragment.js
import JSONSerializer from 'ember-data/serializers/json';

export default JSONSerializer.extend({
  keyForAttribute(attr) {
    return Ember.String.dasherize(attr);
  }
});
```

Could this be generated by default on install, or via an optional json api blueprint ?
